scripts/nn/test/grad_check.dml

```diff
@@ -616,6 +617,114 @@ conv2d_simple = function() {
   }
 }
 
+conv2d_transpose = function() {
+  /*
+   * Gradient check for the 2D convolution transpose layer.
+   */
+  print("Grad checking the 2D convolution transpose layer with L2 loss.")
+
+  N = 1
+  C = 1
+  Hin = 3
+  Win = 3
+  F = 1
+  Hf = 3
+  Wf = 3
+  stride = 2
+  pad = 1
+  out_pad = 1
+
+  X = rand(rows=N, cols=C*Hin*Win)
+
+  [W,b] = conv2d_transpose::init(F, C, Hf, Wf)
+  
+  [out, Hout, Wout] = 
+ 	conv2d_transpose::forward(X, W, b, C, Hin, Win, Hf, Wf, stride, stride, pad, pad, out_pad, out_pad)
+  
+  y = rand(rows=N, cols=F*Hout*Wout)
+  
+  dout = l2_loss::backward(out,y)
+  
+  [dX, dW, db] = 
+  	conv2d_transpose::backward(dout, Hout, Wout, X, W, b, C, Hin, Win, Hf, Wf, stride, stride, pad, pad)
+  
+  h = 1e-5
+  print(" - Grad checking X.")
+  for (i in 1:nrow(X)) {
+    for (j in 1:ncol(X)) {
+      old = as.scalar(X[i,j])
+      X[i,j] = old - h
+
+	  [outmh, Hout, Wout] = 
+	  	conv2d_transpose::forward(X, W, b, C, Hin, Win, Hf, Wf, stride, stride, pad, pad, out_pad, out_pad)
+
+	  lossmh = l2_loss::forward(outmh, y)
+	
+      X[i,j] = old + h
+      [outph, Hout, Wout] = 
+      	conv2d_transpose::forward(X, W, b, C, Hin, Win, Hf, Wf, stride, stride, pad, pad, out_pad, out_pad)
+
+	  lossph = l2_loss::forward(outph, y)
+
+	  X[i,j] = old
+
+	  dX_num = (lossph-lossmh) / (2*h)
+
+	  rel_error = test_util::check_rel_grad_error(as.scalar(dX[i,j]), dX_num, lossph, lossmh)
```

Comment:
Looks like some tabs were accidentally added in this for loop.  Can you replace them each with 2 spaces?  Also, can you please remove the extra new lines?

Same thing for the for loops below.